Added css3 webcolor support for steady colors & random color option

bwu62 opened the pull request:
Added few lines of code in main.py to enable automatic detection of [CSS3 webcolors](https://www.w3.org/TR/SVG11/types.html#ColorKeywords).    
    
Now, instead of `sudo msi-perkeyrgb -s 00ff00` you can simply run `sudo msi-perkeyrgb -s green` or `sudo msi-perkeyrgb -s cornflowerblue` or `sudo msi-perkeyrgb -s peachpuff` or whatever other fancy colors you desire.

The changes use the `name_to_hex` method from the `webcolors` module (which has been included in the `setup.py` as a dependency) to try converting the given `-s` argument from web color to hex. If it fails (e.g. if it's already a hex color), it will revert back to the original behavior and try to set it as a hex color. If that fails, the same original error will be thrown.

Changes have been tested out locally and there appears to be no errors.

bwu62 committed:
added css3 webcolor support for steady colors

bwu62 committed:
slight modification to way webcolor names converted

bwu62 committed:
added random color option; fixed minor issues with webcolors

bwu62 commented:
Fixed a minor issue with webcolor code, updated help text, and added random color option.    
    
Color sampled uniformly randomly in HLS space such that
 - 25% <= luminance < 75%
 - 50% <= saturation < 100%
so that resulting random color won't be too light or dark or desaturated.

hugglesfox commented on msi_perkeyrgb/main.py:
```diff
@@ -126,7 +130,36 @@ def main():
             # If user has requested to display a steady color
             elif args.steady:
                 try:
```
Just an idea but to make things more consistent with other applications, we may want to only try to parse it as a hex if the color is preceded by a `#`, This way instead of using an try block, we could use an if statement to check to check whether it's a hex or a color. This way we can better give feedback to the user whether they typed invalid hex or an invalid color name plus cleans up the code a bit.

bwu62 replied:
Good suggestion but it makes the usage syntax inconsistent with previous versions, which may be a slight nuisance to any existing users that update to the latest version. Do you think it's important enough to change?

hugglesfox replied:
To be honest I don't know and is probably something which @Askannz needs to decide.

bwu62 replied:
Sounds good. Leaving it up to @Askannz for now.

hugglesfox commented on msi_perkeyrgb/main.py:
```diff
@@ -24,7 +27,8 @@ def main():
     parser.add_argument('--list-models', action='store_true', help='List available laptop models.')
-    parser.add_argument('-s', '--steady', action='store', metavar='HEXCOLOR', help='Set all of the keyboard to a steady html color. ex. 00ff00 for green')
+    parser.add_argument('-s', '--steady', action='store', metavar='COLOR', help='Set all of keyboard to a steady hex or css3 color e.g. 00ff00 or green.')
+    parser.add_argument('-r', '--random',action='store_true',help='Set all of keyboard to a random hex color.')
```
Just need some spaces after each `,` to make this PEP8 compliant.

bwu62 committed:
fixed spaces in help text to comply w/ PEP8

bwu62 committed:
added random color print statement; added GL73 to supported devices

bwu62 commented:
^ Forgot to add an item to the commit message so amended it and force-pushed.

Changes:

1. Added print out statement to random color feature so user can see the hex color that was randomly generated.
1. Been using this on my computer (GL73-9SDK-413) for few weeks now and I've had no problems, so I added GL73 to the list of supported devices. It seems to use the same keyboard map as GL63.

bwu62 commented:
Sorry, accidentally hit close button. Request is still open, pending further review or merge.
